app/containers/MessageActions.js

```diff
@@ -208,7 +209,7 @@ class MessageActions extends React.Component {
 		}
 		const deleteOwn = this.isOwn(props);
 		const { Message_AllowDeleting: isDeleteAllowed, Message_AllowDeleting_BlockDeleteInMinutes } = this.props;
-		if (!(this.hasDeletePermission || (isDeleteAllowed && deleteOwn) || this.hasForceDeletePermission)) {
+		if (!(this.hasDeletePermission || (isDeleteAllowed && deleteOwn && this.hasDeleteOwnMsgPermission) || this.hasForceDeletePermission)) {
```

Comment:
Not quite like it.
Check [here](https://github.com/RocketChat/Rocket.Chat/pull/15512/files#diff-b722e88788cc020befe0585d96322a9eR22).
The permission is "Delete own message".
You aren't checking if the message is from the user.
Try to make it as similar to web client as you can, please.

Reply:
@diegolmello is it fine now?